rest/route/agent.go

```diff
@@ -1648,10 +1648,10 @@ func (h *createGitHubDynamicAccessToken) Run(ctx context.Context) gimlet.Respond
 		Permissions: permissions,
 	})
 	if err != nil {
-		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "creating installation token for '%s/%s'", h.owner, h.repo))
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "creating installation token for '%s/%s'", h.owner, h.repo))
```

Comment:
We can't change this yet because of #8704 - the existing GitHub app client doesn't retry internally in all the cases where it needs to. Can we revert it back?

Reply:
That sounds good to me!

Reply:
I didn't revert the check for token = "" since i don't think retries would help, what do you think?

Reply:
I see in your PR you did all the github ones, so I'm going to revert them all in this one

Reply:
That seems reasonable to me! The only way that could really happen is if GitHub returns a success response but no token, which seems like one of those "should never happen" scenarios.